apps/bmd/src/AppRoot.tsx

```diff
@@ -141,898 +147,1091 @@ export const activationStorageKey = 'activation'
-          // Trigger application of “See More” buttons based upon scroll-port.
-          window.dispatchEvent(new Event('resize'))
-        }
+    case 'processVoterCard':
+      return {
+        ...state,
+        ...initialCardState,
+        ...action.voterState,
+        isVoterCardExpired:
+          state.voterCardCreatedAt === 0 &&
+          utcTimestamp() >=
+            action.voterState.voterCardCreatedAt! +
+              GLOBALS.CARD_EXPIRATION_SECONDS,
       }
-    )
-  }
-
-  public setDocumentFontSize = (textSize: number = GLOBALS.TEXT_SIZE): void => {
-    document.documentElement.style.fontSize = `${GLOBALS.FONT_SIZES[textSize]}px`
-  }
-
-  public updateAppPrecinctId = (appPrecinctId: string): void => {
-    this.setState({ appPrecinctId }, this.resetTally)
+    case 'pauseCardProcessing':
+      return {
+        ...state,
+        pauseProcessingUntilNoCardPresent: true,
+      }
+    case 'resumeCardProcessing':
+      return {
+        ...state,
+        pauseProcessingUntilNoCardPresent: false,
+      }
+    case 'setMachineConfig':
+      return {
+        ...state,
+        machineConfig:
+          action.machineConfig || initialHardwareState.machineConfig,
+      }
+    case 'updateLastVoteUpdateAt':
+      return {
+        ...state,
+        lastVoteUpdateAt: action.date,
+      }
+    case 'unconfigure':
+      return {
+        ...state,
+        ...initialUserState,
+      }
+    case 'updateVote': {
+      return {
+        ...state,
+        votes: {
+          ...state.votes,
+          [action.contestId]: action.vote,
+        },
+      }
+    }
+    case 'forceSaveVoteFlag':
+      return {
+        ...state,
+        forceSaveVoteFlag: action.value,
+      }
+    case 'resetBallot':
+      return {
+        ...state,
+        ...initialCardState,
+        ...initialVoterState,
+      }
+    case 'setUserSettings':
+      /* istanbul ignore next */
+      if (Object.keys(action.userSettings).join(',') !== 'textSize') {
+        throw new Error('unknown userSetting key')
+      }
+      if (action.userSettings.textSize === state.userSettings.textSize) {
+        return state
+      }
+      return {
+        ...state,
+        userSettings: {
+          ...state.userSettings,
+          ...action.userSettings,
+        },
+      }
+    case 'updateAppPrecinctId':
+      return {
+        ...state,
+        ...resetTally,
+        appPrecinctId: action.appPrecinctId,
+      }
+    case 'enableLiveMode':
+      return {
+        ...state,
+        ...resetTally,
+        isLiveMode: true,
+        isPollsOpen: initialAppState.isPollsOpen,
+      }
+    case 'toggleLiveMode':
+      return {
+        ...state,
+        ...resetTally,
+        isLiveMode: !state.isLiveMode,
+        isPollsOpen: initialAppState.isPollsOpen,
+      }
+    case 'togglePollsOpen':
+      return {
+        ...state,
+        isPollsOpen: !state.isPollsOpen,
+      }
+    case 'updateTally': {
```

Comment:
This name is confusing for what it does. It doesn't change the tally, does it? It just increments the `ballotsPrintedCount`.

Reply:
But it both increments `ballotsPrintedCount` and updates `tally`:

![image](https://user-images.githubusercontent.com/28444/104227421-37ecda80-53fe-11eb-96c8-f9eee2e3f733.png)